libloki/proof-of-work.js

```diff
@@ -1,67 +1,102 @@
 const hash = require('js-sha512');
 const bb = require('bytebuffer');
 
+const NONCE_LEN = 8;
+
 // Increment Uint8Array nonce by 1 with carrying
 function incrementNonce(nonce) {
-  let idx = nonce.length - 1;
-  nonce[idx] += 1;
-  // Nonce will just reset to 0 if all values are 255 causing infinite loop, should never happen
-  while (nonce[idx] == 0 && idx >= 0) {
-    nonce[--idx] += 1;
+  let idx = NONCE_LEN - 1;
+  const newNonce = nonce;
+  newNonce[idx] += 1;
+  // Nonce will just reset to 0 if all values are 255 causing infinite loop
+  while (nonce[idx] === 0 && idx > 0) {
+    idx -= 1;
+    newNonce[idx] += 1;
   }
   return nonce;
 }
 
 // Convert a Uint8Array to a base64 string
 function bufferToBase64(buf) {
-  let binstr = Array.prototype.map.call(buf, function (ch) {
+  function mapFn(ch) {
       return String.fromCharCode(ch);
-  }).join('');
-  return bb.btoa(binstr);
+  };
+  const binaryString = Array.prototype.map.call(buf, mapFn).join('');
+  return bb.btoa(binaryString);
```

Comment:
I guess this could be also a one-liner:

`return bb.btoa(Array.from(buf).map(x => String.fromCharCode(x)).join(''));`